src/Stratis.Bitcoin.Features.Wallet/AddressBookManager.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Linq;
+using Microsoft.Extensions.Logging;
+using Stratis.Bitcoin.Configuration;
+using Stratis.Bitcoin.Features.Wallet.Interfaces;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Wallet
+{
+    public class AddressBookManager : IAddressBookManager
+    {
+        /// <summary>File extension for wallet files.</summary>
+        private const string AddressBookFileName = "addressbook.json";
+
+        /// <summary>Instance logger.</summary>
+        private readonly ILogger logger;
+
+        /// <summary>An object capable of storing <see cref="Wallet"/>s to the file system.</summary>
+        private readonly FileStorage<AddressBook> fileStorage;
+
+        private readonly AddressBook addressBook;
+
+        public AddressBookManager(
+            ILoggerFactory loggerFactory,
+            DataFolder dataFolder)
+        {
+            Guard.NotNull(loggerFactory, nameof(loggerFactory));
+            Guard.NotNull(dataFolder, nameof(dataFolder));
+
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+            this.fileStorage = new FileStorage<AddressBook>(dataFolder.RootPath);
+
+            if (this.fileStorage.Exists(AddressBookFileName))
+            {
+                this.addressBook = this.fileStorage.LoadByFileName(AddressBookFileName);
+            }
+            else
+            {
+                this.addressBook = new AddressBook();
+                this.fileStorage.SaveToFile(this.addressBook, AddressBookFileName);
```

Comment:
We should probably not save to disk on constructors, this is called form the DI and we have no control over when constructors are called and they should also be as light as possible.

Reply:
This will be done only the first time this code is run though.
Maybe in a start method?